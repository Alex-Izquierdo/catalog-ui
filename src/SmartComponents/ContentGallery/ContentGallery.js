import React from 'react';
import propTypes from 'prop-types';
import { Section } from '@red-hat-insights/insights-frontend-components';
import { CardLoader } from '../../PresentationalComponents/Shared/LoaderPlaceholders';
import { Bullseye, Text, TextVariants } from '@patternfly/react-core';

const NoItems = () => (
  <Bullseye>
    <Text component={ TextVariants.h1 }>No items found</Text>
  </Bullseye>
);

const ContentGallery = ({ isLoading, items }) => isLoading ? <CardLoader /> : (
  <Section type="content">
    <div className="content-gallery" >
      { items.length > 0 ? items : <NoItems /> }
    </div>
  </Section>
);

ContentGallery.propTypes = {
  isLoading: propTypes.bool,
  items: propTypes.array
};
export default ContentGallery;
