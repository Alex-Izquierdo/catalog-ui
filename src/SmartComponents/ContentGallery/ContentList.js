import React, { Component } from 'react';
import { BarLoader } from 'react-spinners';
import propTypes from 'prop-types';
import { PageHeader, PageHeaderTitle, Table, Section } from '@red-hat-insights/insights-frontend-components';
import { Bullseye } from '@patternfly/react-core';

const defaultProperty = property => {
    return [].includes(property);
};

const propDetails = item => {
    let details = {};

    for (let property in item) {
        if (item.hasOwnProperty(property) && !defaultProperty(property)) {
            if (item[property] && item[property] !== undefined) {
                details[property] = item[property].toString();
            }
        }
    }

    return details;
};

const itemDetails = props => {
    let details = propDetails(props);
    return (
        <React.Fragment>
            <div>{ details }</div>
        </React.Fragment>
    );
};

class ContentList extends Component {
    render() {
        if (this.props.isLoading || (this.props.items && this.props.items.length) > 0) {
            return (
                <React.Fragment>
                    <br />
                    <Bullseye>
                        <BarLoader color={ '#00b9e4' } loading={ this.props.isLoading } />
                    </Bullseye>
                    <Section type='content'>
                        { (this.props.items && this.props.items.length > 0) && (
                            <Table
                                header={ Object.keys(this.props.items[0]) }
                                rows={ this.props.items.map(item => { let row = {}; row.cells = (Object.values(item)).map(val => {return val === undefined ? '' : val.toString();}); return row;}) }/>)
                        }
                    </Section>
                </React.Fragment>
            );
        }
        else if (!this.props.isLoading) {
            return (
                <React.Fragment>
                    <div>
                        <PageHeader>
                            <PageHeaderTitle title={ 'No Orders' }/>
                        </PageHeader>
                    </div>
                </React.Fragment>
            );
        }
    }
}

ContentList.propTypes = {
    isLoading: propTypes.bool,
    items: propTypes.array,
    pageSize: propTypes.number,
    fetchData: propTypes.func
};
export default ContentList;
