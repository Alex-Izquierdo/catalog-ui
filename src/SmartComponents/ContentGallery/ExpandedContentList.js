import React, { Component } from 'react';
import { BarLoader } from 'react-spinners';
import propTypes from 'prop-types';
import { PageHeader, PageHeaderTitle, Table, Section } from '@red-hat-insights/insights-frontend-components';
import { Bullseye } from '@patternfly/react-core';
import { Row, Col, ListView,
    ListViewGroupItem,
    ListViewGroupItemHeader,
    ListViewMainInfo,
    ListViewLeft,
    ListViewActions,
    ListViewBody,
    ListViewIcon,
    ListViewDescriptionHeading,
    ListViewDescriptionText,
    ListViewExpand,
    ListViewCheckBox,
    ListViewDescription,
    ListViewGroupItemContainer,
    ListViewInfoItem,
    ListViewItem,
    ListViewAdditionalInfo } from 'patternfly-react';

const defaultProperty = property => {
    return [].includes(property);
};

const propDetails = item => {
    let details = {};

    for (let property in item) {
        if (item.hasOwnProperty(property) && !defaultProperty(property)) {
            if (item[property] && item[property] !== undefined) {
                details[property] = item[property].toString();
            }
        }
    }

    return details;
};

const itemDetails = (item) => {
    let row = {};
    row.cells = (Object.values(item)).map(val => {return val === undefined ? '' : val.toString();});
    return (
        <ListViewItem
            actions={ <div /> }
            checkboxInput={ <input /> }
            leftContent={ <ListViewIcon /> }
            heading="Item 1"
            description="This is Item 1 description"
        >
            <Row>
                <Col sm={ 11 }>
                    { row }
                </Col>
            </Row>
        </ListViewItem>);
};

class ExpandedContentList extends Component {
    render() {
        if (this.props.isLoading || (this.props.items && this.props.items.length) > 0) {
            return (
                <React.Fragment>
                    <br />
                    <Bullseye>
                        <BarLoader color={ '#00b9e4' } loading={ this.props.isLoading } />
                    </Bullseye>
                    <Section type='content'>
                        { (this.props.items && this.props.items.length > 0) && (
                            <ListView>
                                header={ Object.keys(this.props.items[0]) }
                                { this.props.items.map(item => {itemDetails(item);}) }
                            </ListView>)
                        }
                    </Section>
                </React.Fragment>
            );
        }
        else if (!this.props.isLoading) {
            return (
                <React.Fragment>
                    <div>
                        <PageHeader>
                            <PageHeaderTitle title={ 'No Orders' }/>
                        </PageHeader>
                    </div>
                </React.Fragment>
            );
        }
    }
}

ExpandedContentList.propTypes = {
    isLoading: propTypes.bool,
    items: propTypes.array,
    pageSize: propTypes.number,
    fetchData: propTypes.func
};
export default ExpandedContentList;
