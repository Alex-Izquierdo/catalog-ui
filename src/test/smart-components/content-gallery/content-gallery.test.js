import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ContentGallery from '../../../smart-components/content-gallery/content-gallery';

describe('<ContentGallery />', () => {
  it('should render in loading state', () => {
    expect(toJson(shallow(<ContentGallery isLoading />))).toMatchSnapshot();
  });

  it('should render no items placeholder', () => {
    const props = {
      isLoading: false,
      items: []
    };

    const wrapper = mount(<ContentGallery {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render all items', () => {
    const props = {
      isLoading: false,
      items: [1, 2, 3].map((item) => <div key={item}>{item}</div>)
    };

    const wrapper = mount(<ContentGallery {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
