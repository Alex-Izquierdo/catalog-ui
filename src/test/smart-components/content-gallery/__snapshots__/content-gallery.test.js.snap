// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentGallery /> should render all items 1`] = `
<ContentGallery
  isLoading={false}
  items={
    Array [
      <div>
        1
      </div>,
      <div>
        2
      </div>,
      <div>
        3
      </div>,
    ]
  }
>
  <Section
    type="content"
  >
    <section
      className="ins-l-content"
    >
       
      <div
        className="content-gallery"
      >
        <div
          key="1"
        >
          1
        </div>
        <div
          key="2"
        >
          2
        </div>
        <div
          key="3"
        >
          3
        </div>
      </div>
       
    </section>
  </Section>
</ContentGallery>
`;

exports[`<ContentGallery /> should render in loading state 1`] = `
<CardLoader
  items={5}
/>
`;

exports[`<ContentGallery /> should render no items placeholder 1`] = `
<ContentGallery
  isLoading={false}
  items={Array []}
>
  <Section
    type="content"
  >
    <section
      className="ins-l-content"
    >
       
      <div
        className="content-gallery"
      >
        <NoItems>
          <Bullseye
            className=""
            component="div"
          >
            <div
              className="pf-l-bullseye"
            >
              <Text
                className=""
                component="h1"
              >
                <h1
                  className=""
                  data-pf-content={true}
                >
                  No items found
                </h1>
              </Text>
            </div>
          </Bullseye>
        </NoItems>
      </div>
       
    </section>
  </Section>
</ContentGallery>
`;
